infra/build/developer-tools/build/scripts/task_helper_functions.sh

```diff
@@ -384,7 +384,22 @@ function check_headers() {
   echo "Checking file headers"
   # Use the exclusion behavior of find_files
   find_files . -type f -print0 \
-    | compat_xargs -0 python /usr/local/verify_boilerplate/verify_boilerplate.py
+    | compat_xargs -0 python /usr/local/verify_boilerplate/verify_boilerplate.py 2>&1 \
+    | grep -v 'have incorrect boilerplate headers' > legacy_check.log
+  if [[ -v EXCLUDE_HEADER_CHECK ]]; then
```

Comment:
I am not sure if I have to introduce **_EXCLUDE_HEADER_CHECK_** to other places or do any other updates to make it working like with [EXCLUDE_LINT_DIRS](https://github.com/terraform-google-modules/terraform-example-foundation/blob/1fad10b2af93069a7369cd100a3fcf2cddaa6f74/build/lint.cloudbuild.yaml#L23)

Reply:
Do we need to add a separate exclude for this? I'd prefer to continue using `EXCLUDE_LINT_DIRS`.

Reply:
I'm not sure how its (`EXCLUDE_LINT_DIRS`) expected usage is. My thinking was that with `EXCLUDE_LINT_DIRS ` we cannot exclude specific files - but only whole directories. Also can we provide a list of comma separated options? And the other thing is, what if we wanted to only skip `header_checks` and not all the other linting. These were the things I was taking into consideration. WDYT?

Reply:
Correct, `EXCLUDE_LINT_DIRS` only works for directories. So far, that has been sufficient.

I don't think we have any cases of only wanting to exclude a single file from header checks. Given that, I suggest we don't include it unless/until we have a specific use case.

Reply:
So I am working on [this PR](https://github.com/GoogleCloudPlatform/anthos-samples/pull/59) which has a `cloud-config` file to be passed into the cloud-init system. If this file has any headers the cloud-init fails. I have to just skip only this file. So this is the case I am trying to address here. 

For now, I have to evade it by creating a directory to only hold this file and point to it in `EXCLUDE_LINT_DIRS`. Ideally, I wouldn't want to do this and I want the other linting to run on that file. Do you think this is a valid use case?

Reply:
Got it. Since you have a use case, I'm fine to include it.

However, I do suggest we use a generalized approach with find_files instead of relying on specific exclusion logic for `addlicense` specifically, particularly because the skip argument seems to be [for extensions](https://github.com/google/addlicense/blob/master/main.go#L77).

Reply:
Thank you. 

Yes, looking into what `find_files` does. The `addlicense` skip is not properly documented. It works for specific files too. I tested it out.

Reply:
Resolving this thread. Please re-open if need be.

Reply:
To be clear, I was hoping we would move the `EXCLUDE_HEADER_CHECK` logic into `find_files` so it's consistent (instead of relying on specific `-skip` flag in `addlicense`. It's a minor point but I'd rather have our exclusion logic centralized and tool-agnostic.

Reply:
All exclusion logic moved to find_files in https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/954/commits/268a08357246a0794b0b24ac274a48c400a4874f